staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd2.go

```diff
@@ -39,10 +42,30 @@ func newETCD2Storage(c storagebackend.Config) (storage.Interface, DestroyFunc, e
 	if err != nil {
 		return nil, nil, err
 	}
+
+	stopCh := make(chan struct{})
+	if c.AutoSyncInterval > 0 {
+		go wait.Until(func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			err := client.Sync(ctx)
+			cancel()
+			if err != nil {
+				glog.Warningf("etcd auto sync endpoints failed: %v", err)
+				return
+			}
+		}, c.AutoSyncInterval, stopCh)
+	}
+
 	s := etcd.NewEtcdStorage(client, c.Codec, c.Prefix, c.Quorum, c.DeserializationCacheSize, etcd.IdentityTransformer)
-	return s, tr.CloseIdleConnections, nil
+	destroyFunc := func() {
+		tr.CloseIdleConnections()
+		close(stopCh)
```

Comment:
close stopCh first, to avoid a resync reopening a connection right after you close it?

Reply:
make sense